### ISSUE TITLE
Is having the same name for commits considered good practice?

### ISSUE THREAD

**Issue by kwmiw** [nus-cs2103-AY2021S2/forum#166]
Whenever I update my automated testing files to suit the newest commits, I use the same commit name for each level - 'Update Automated UI Testing'. Is this considered confusing or generally understood?

**Comment by YuFeng0930** [comment]
Personally, I think it's fine for "Automated UI Testing", but maybe you still can give more detailed information (e.g. what update did you make) after the same commit name.

**Comment by vevek** [comment]
Referring to [this link](https://se-education.org/guides/conventions/git.html), 

You can add more info into your commit message subject to differentiate between commits. You can also specify more information as necessary within the commit message body. There are some really useful examples in the link above!

**Comment by samuelfangjw** [comment]
In my opinion, I think it is fine to have a commit message with the same name as long as what the change is and why the change was made is clear. For example, a message such as "Fix typo in user guide" makes it very clear that we are fixing a typo because there was a mistake. 

However, I feel that "Update Automated UI Testing" does not clearly show why the change was made. In this case perhaps the context behind the change (e.g. to test for the list command) should be added to the body or the subject title so other users who are reviewing the code will get a better idea of why the commit was done without having to look through at the diffs page.

**Comment by damithc** [comment]
> In my opinion, I think it is fine to have a commit message with the same name as long as what the change is and why the change was made is clear. For example, a message such as "Fix typo in user guide" makes it very clear that we are fixing a typo because there was a mistake.

That's true. In addition, there is _some_ value in making the subject unique (and more informative), at least among the recent commits. e.g.,
Consider three consecutive commits:
```
UG: Fix typo teh -> the
UG: Fix typo yyes -> yes
UG: Fix typos red -> read etc.
```
vs the same three commits:
```
UG: Fix typo
UG: Fix typo
UG: Fix typos
```

**Comment by rajobasu** [comment]
I believe one way to tackle might be to squash similar consecutive commits together like in @damithc ’s example?

**Comment by damithc** [comment]
> I believe one way to tackle might be to squash similar consecutive commits together like in @damithc ’s example?

Yes, can squash if those commits haven't been pushed to the remote yet and those commits actually do the same thing (i.e., if they do different things, they should be kept as separate commits).
